Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -395,7 +389,7 @@ public final class PackageBuilder {
 
             // Diagnose broken symlinks.
             if fileSystem.isSymlink(path) {
-                diagnostics.emit(.brokenSymlink(path), location: self.diagnosticLocation)
+                self.diagnosticsEmitter.emit(.brokenSymlink(path))
```

Comment:
Here is seems as if we're losing some information.  Or is the context of the package handled through the emitter?  (I might have missed it but will take another look).  If so that would be great, since that is context that should not have to be added individually to each diagnostic.

Reply:
Oh right, I found the line of code, it was just a little ways up in the file.  That's great!

Reply:
right, metadata can be specified at the emitter level or the individual diagnostic level and the two get merged if both are specified

Reply:
Great.  That seems flexible.  It relates to the discussion of location vs context, above, of course.  Here I could see the emitter usually having the context (what task was going on) and the individual diagnostic usually having the location (e.g. specific line in the manifest or code).